### PR TITLE
server: Update log format for server

### PR DESCRIPTION
**Pull request by MyonKeminta**
Signed-off-by: MyonKeminta <MyonKeminta@users.noreply.github.com>

## What have you changed? (mandatory)

This PR updates log format for `server/` directory of the source code.
`server/service`  directory is not updated in this PR. It will be done in another PR.

## What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:

- Engineering (engineering change which doesn't change any feature or fix any issue)

## How has this PR been tested? (mandatory)

By CI.

## Does this PR affect documentation (docs) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

### CODE REVIEW

File: src/server/config.rs [hunk 1]
```diff
@@ -162,8 +162,8 @@ impl Config {
             box_try!(config::check_addr(&self.advertise_addr));
         } else {
             info!(
-                "no advertise-addr is specified, falling back to {}.",
-                self.addr
+                "no advertise-addr is specified, falling back to default addr";
+                "fall_back_to" => %self.addr
```
Comment (breeswish) [inline src/server/config.rs]: fall_back_addr

File: src/server/config.rs [hunk 2]
```diff
@@ -162,8 +162,8 @@ impl Config {
             box_try!(config::check_addr(&self.advertise_addr));
         } else {
             info!(
-                "no advertise-addr is specified, falling back to {}.",
-                self.addr
+                "no advertise-addr is specified, falling back to default addr";
+                "fall_back_addr" => %self.addr
```
Comment (zhangjinpeng1987) [inline src/server/config.rs]: I think addr is enough.

File: src/server/transport.rs [hunk 3]
```diff
@@ -218,14 +218,15 @@ impl<T: RaftStoreRouter + 'static, S: StoreAddrResolver + 'static> ServerTranspo
                 .inc();
             // If we are resolving the address, drop the message here.
             debug!(
-                "store {} address is being resolved, drop msg {:?}",
-                store_id, msg
+                "store address is being resolved, msg dropped";
+                "store_id" => store_id,
+                "msg" => ?msg
```
Comment (zhangjinpeng1987) [inline src/server/transport.rs]: s/msg/message. I prefer use full name for short tag.

File: src/server/transport.rs [hunk 4]
```diff
@@ -343,8 +344,11 @@ impl<T: RaftStoreRouter + 'static, S: StoreAddrResolver + 'static> ServerTranspo
 
         if let Err(e) = self.raft_router.report_unreachable(region_id, to_peer_id) {
             error!(
-                "report peer {} on store {} unreachable for region {} failed {:?}",
-                to_peer_id, store_id, region_id, e
+                "report peer unreachable failed";
+                "peer_id" => to_peer_id,
```
Comment (zhangjinpeng1987) [inline src/server/transport.rs]: I prefer the order store > region > peer.

Comment (zhangjinpeng1987) [reply src/server/transport.rs]: s/peer_id/to_peer_id
s/store_id/to_store_id

File: src/server/transport.rs [hunk 5]
```diff
@@ -396,8 +402,11 @@ impl<T: RaftStoreRouter + 'static> SnapshotReporter<T> {
                 .report_snapshot_status(self.region_id, self.to_peer_id, status)
         {
             error!(
-                "report snapshot to peer {} in store {} with region {} err {:?}",
-                self.to_peer_id, self.to_store_id, self.region_id, e
+                "report snapshot to peer failes";
+                "to_peer_id" => self.to_peer_id,
+                "store_id" => self.to_store_id,
```
Comment (zhangjinpeng1987) [inline src/server/transport.rs]: s/store_id/to_store_id

**Comment by MyonKeminta** [comment]
@zhangjinpeng1987 @breeswish PTAL again thanks!

**Pull request merged**